### PR TITLE
chore(linux): allow to trigger Jenkins build from script

### PR DESCRIPTION
**Pull request by ermshiperete**
This change also refactors `triggerBuilds` to use the definitions
from `trigger-definitions.config`. Jenkins configurations are
identified by the suffix `_Jenkins`.

---
This change is [Reviewable](https://reviewable.io/reviews/keymanapp/keyman/2697)

### CODE REVIEW

File: resources/build/trigger-builds.sh [hunk 1]
```diff
@@ -6,22 +6,24 @@
...
+  eval TEAMCITY_VCS_ID='${'vcs_$base'}'
+  echo "base=$base, TEAMCITY_VCS_ID=${TEAMCITY_VCS_ID}"
 
-  for TEAMCITY_BUILDTYPE in "${TEAMCITY_BUILDTYPES[@]}"; do
-    triggerBuild $TEAMCITY_BUILDTYPE $TEAMCITY_VCS_ID
+  for platform in ${available_platforms[@]}; do
+    eval builds='(${'bc_${base}_${platform}'[@]})'
+    for build in "${builds[@]}"; do
+      if [[ $build == "" ]]; then continue; fi
+      if [ "${build:(-8)}" == "_Jenkins" ]; thend
```
Comment (mcdurdin) [inline resources/build/trigger-builds.sh]:
```suggestion
      if [ "${build:(-8)}" == "_Jenkins" ]; then
```

Comment (ermshiperete) [reply resources/build/trigger-builds.sh]: What!? How did that happen? I tested it before committing! :smiley:

Comment (mcdurdin) [reply resources/build/trigger-builds.sh]: It's also quite common to see `:wq!` in scripts. 🤣

**Comment by mcdurdin** [comment]
This failed with 

```
[08:16:14]Step 2/2: Trigger test builds (Windows) (Command Line) (5s)
[08:16:14][Step 2/2] Starting: C:\BuildAgent\temp\agentTmp\custom_script4823804673406246571.cmd
[08:16:14][Step 2/2] in directory: C:\BuildAgent\work\99b311828f4ee7c\keyman
[08:16:16][Step 2/2] . Get information about pull request #2697 from GitHub
[08:16:16][Step 2/2] . Ensure our local branch is up to date
[08:16:18][Step 2/2] . Get list of changed files in the pull request
[08:16:18][Step 2/2] . Find platforms that have changes
[08:16:18][Step 2/2] . Start test builds
[08:16:18][Step 2/2] # all: changes detected
[08:16:18][Step 2/2] # android: changes detected
[08:16:18][Step 2/2]   -- Triggering build configuration KeymanAndroid_TestPullRequests on teamcity
[08:16:18][Step 2/2] {"id":184090,"buildTypeId":"KeymanAndroid_TestPullRequests","state":"queued", ...
[08:16:18][Step 2/2]   -- Triggering build configuration KeymanAndroid_TestSamplesAndTestProjects on teamcity
[08:16:19][Step 2/2] {"id":184091,"buildTypeId":"KeymanAndroid_TestSamplesAndTestProjects","state":"queued", ...
[08:16:19][Step 2/2] # ios: changes detected
[08:16:19][Step 2/2]   -- Triggering build configuration Keyman_iOS_TestPullRequests on teamcity
[08:16:19][Step 2/2] {"id":184092,"buildTypeId":"Keyman_iOS_TestPullRequests","state":"queued", ...
[08:16:19][Step 2/2]   -- Triggering build configuration Keyman_iOS_TestSamplesAndTestProjects on teamcity
[08:16:19][Step 2/2] {"id":184093,"buildTypeId":"Keyman_iOS_TestSamplesAndTestProjects","state":"queued", ...
[08:16:19][Step 2/2] # linux: changes detected
[08:16:19][Step 2/2]   -- Triggering build configuration KeymanLinux_TestPullRequests on teamcity
[08:16:19][Step 2/2] {"id":184094,"buildTypeId":"KeymanLinux_TestPullRequests","state":"queued", ...
[08:16:19][Step 2/2]   -- Triggering build configuration Keyman_Common_KPAPI_TestPullRequests_Linux on teamcity
[08:16:19][Step 2/2] {"id":184095,"buildTypeId":"Keyman_Common_KPAPI_TestPullRequests_Linux","state":"queued", ...
[08:16:19][Step 2/2]   -- Triggering build configuration pipeline-keyman-packaging/2697 on Jenkins
[08:16:19][Step 2/2] /c/BuildAgent/work/99b311828f4ee7c/keyman/resources/build/trigger-builds.sh: line 61: $3: unbound variable
[08:16:19][Step 2/2] Process exited with code 1
[08:16:19][Step 2/2] Process exited with code 1 (Step: Trigger test builds (Windows) (Command Line))
[08:16:20][Step 2/2] Step Trigger test builds (Windows) (Command Line) failed
```

**Comment by darcywong00** [comment]
I'll go ahead and merge cause I plan to use Jenkins today

**Pull request merged**